### ISSUE TITLE
Manual approval between plan and apply

### ISSUE THREAD

**Issue by brettcurtis** [hashicorp/learn-terraform-github-actions#24]
Has anyone found a decent way to have some sort of manual approval process between plan and apply? Between "steps" is really what I'm after. But I suppose it could even be between jobs. A job to run plan and approve followed by a job to run apply.

**Comment by msavita-cloud** [comment]
HELLLLLOOOOO

**Comment by bansawbanchee** [comment]
look into the `workflow_dispatch` event. https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/ I personally use this for a few manually triggered jobs.

**Comment by brettcurtis** [comment]
We are attempting to using the new "requested reviewers" process for this now in Actions. Still have some things to figure out regarding the flow of it all. If it's useful I can share an actions file when we are done.

**Comment by TrongTheAlpaca** [comment]
@brettcurtis I was wondering if you arrived to any interesting conclusion about the idea using "requested reviewers" for this?

**Comment by brettcurtis** [comment]
Yeah, we ended up going all in on GitHub here. We use [environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) and [required reviewers](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#required-reviewers) like I mention above. We also use [called workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) in an effort to be as DRY as possible. I can share our called workflows if that would be helpful. We have a called workflow for each of the major cloud providers, GCP, AWS & Azure.

**Comment by brettcurtis** [comment]
@TrongTheAlpaca here you go, very specific to how I use Terraform but hopefully it can give you some ideas: https://github.com/lzysh/github-terraform-called-workflows/blob/main/.github/workflows/aws-plan-and-apply-called.yml

Gives us something like this (this is a pic of Google workflow but AWS is similar) allows us to have human approval between each plan and apply job:
![image](https://user-images.githubusercontent.com/1610100/161629479-a0c49912-4cff-4823-920e-b1e0db992159.png)
